### DISCUSSION TITLE
Feature Request - multiple glucose levels/settings

### DISCUSSION THREAD

**Discussion by bryanw20** [NightscoutFoundation/xDrip#4096] [Ideas]
Not sure what to call it; might have to be alert levels; but maybe more a T2 than T1's add more levels, then we can have more colors on the graph.
Example: 
Low  - set  at <80 color could be Yellow
Normal/in range - set to 80-120 - Green
High (med high, ???) set to 120-150/180 whatever - Orange
High - 150/180+ - red

Is this possible?
If it is possible; would it carry over to watch faces? {Pebble for me}

**Comment by jwoglom** [comment]
You can create custom alarms for these levels (in Alarms and Alerts->Alerts List) but I don't think more than the 3 ranges (below, in, and above range) are supported on the graph. This could potentially be done as an expansion of the "Glucose Units" preferences page, which currently has just a high and low glucose value option. I'm not sure if those two values are used in parts of the application other than the graph, however; if they are not, then I could see some value to having a UI like the Alerts List in what is currently Glucose Units (although if it became more significant, it should probably be renamed or spun off in a settings pane that specifically refers to the graph). I could see there being problems with this if code throughout the application besides the alerts depends on there being globally set HIGH and LOW variables. @jamorham, would this be possible with the current architecture?

**Comment by Navid200** [comment]
@bryanw20 Would you please close this issue unless it has not been addressed?
Thanks

**Comment by bryanw20** [comment]
> @bryanw20 Would you please close this issue unless it has not been addressed?
> Thanks

as a t2 this would be cool; but apparently not an option.

**Comment by Navid200** [comment]
@bryanw20 Sorry, I didn't mean that this cannot be implemented.  I thought with the changes since then it was already possible.  
Reading your post again, I see I completely misunderstood.  And, not that your description was at fault.

I know that you can set alarms at different levels.  But, you want the color of the dots to be more than just the existing 3 colors.
I am sorry.  You are more than welcome to re-open the issue.

**Comment by bryanw20** [comment]
> @bryanw20 Sorry, I didn't mean that this cannot be implemented. I thought with the changes since then it was already possible.
> Reading your post again, I see I completely misunderstood. And, not that your description was at fault.
> 
> I know that you can set alarms at different levels. But, you want the color of the dots to be more than just the existing 3 colors.
> I am sorry. You are more than welcome to re-open the issue.

It's fine, obviously it's been years, guess I'm the only one that wanted it that way.

Basically a low, in range, high and very high.

**Comment by Navid200** [comment]
I see nothing wrong with that request.   I'm not one of the developers.  But, I see nothing wrong with leaving this request open.  I misunderstood.  And I will happily add a label to it to show that it is an enhancement request.

**Comment by tolot27** [comment]
@bryanw20 You can have as many alarms as you like - each with a different BG threshold. What's is the added value having a finer-grained graph segmentation except visual coloring? From the medical point of view there is no advantage. The Time In Range (TIR) is solely calculated based on the hypo/hyper thresholds.

**Comment by Navid200** [comment]
@tolot27 That's not the request.
The request is to represent the blood glucose graphically with more than the current 3 colors.  Right now, we see all the in range measurements in blue, all the lows in red, and all the highs in orange.

This is a request.  We may disagree with the priority for implementing it.  But, it is a valid request.  There are so many request that re open.  Why shouldn't this?

**Comment by tolot27** [comment]
@Navid200 I think I understand the request, even if it is not precise enough:

@bryanw20 requested:
> add more levels, then we can have more colors on the graph.

He did not mention whether he just want additional threshold lines and/or coloring the BG values according to a certain threshold. In any way, he wants to have additional bins of the BG value range (mathematically speaking) visually separated. That's what I read from his request.

I'm not against reopening this issue. My concern is about the added value from the medical point of view. Also, it should not irritate the users. Most of us expect two thresholds. Making it configurable adds more settings to the already long list of preferences. 
IMHO, only a very small user group, if not just a single user, will have a benefit of such a feature.

**Comment by Navid200** [comment]
I am closing this for now.  However, we will consider it when the number of open issues has decreased.
In the meantime, I need the following information about what will make you happy.

1- The two new color possibility available on the phone only but not available in statistics
2- The two new colors available on the phone only and also included in the statistics

We need to know if this is going to be useless to you if it is not also added to the statistics, which will make it a more demanding change.

**Comment by Navid200** [comment]
I would like to address this.  But, I still have not got to it.
I am going to convert this to a discussion.  It will still be found through search.  And developers will still be able to see it.

Thanks
